### ISSUE TITLE
Workflow 3: similarity

### ISSUE THREAD

**Issue by cbizon** [NCATS-Gamma/robokop#180]
The first module of workflow 1 contains similarity as a step in a module.  In particular, we're looking for phenotypes that are similar to other phenotypes.  The definition of similarity is unspecified.  

How do we do this?

1. We have a similarity function, we could do e.g. similarity of phenotype by disease?  But how do we incorporate this into a question?

2. We have sets, but does this do the same thing?

3. I was hoping to use HP to find things that are nearby in the ontology, but this is a mistake: often HP will model things by having an 'abnormal X', with 'increased X' and 'decreased X' as children.  For this question, similarity would be more like finding 'increased X' and 'increased Y' where X and Y are both types of blood cells.  Is there a way to do it from the ontology itself?

**Comment by patrickkwang** [comment]
To do similarity by disease, I would make a robokop query like this: (:phenotype {id:XXX})--[(:disease)]--(:phenotype) where the brackets indicate a set. When the results are ranked, it will prefer the ones where the set is larger (the phenotypes that share more diseases with the named one).

**Comment by cbizon** [comment]
So in this case, any publication counts from disease to phenotype are not used?

**Comment by kennethmorton** [comment]
No. They would definitely be used. Are you suggesting that the shouldn’t be? Or to more faithfully do “similarity” they shouldn’t be?

**Comment by patrickkwang** [comment]
Publication counts are used. I should have been more specific. All other things being equal, larger sets are preferred in this case. But diseases connected to the phenotypes with greater weight (more publications) are worth more. There are some ugly details in how publications counts are converted to admittances in our latest EE-inspired ranking scheme.

**Comment by cbizon** [comment]
There's nothing necessarily wrong with using the papers there, but it does present a question of what's the right balance between weighting particular connections vs allowing the counts to influence stuff.  If I understand how the ranker is working, one very well supported P-D-P' path essentially short circuits the network, and at that point, it may not matter how many D's a different P' has in common with P.  So there's a tuning parameter there that classical similarity sets one way and we have set somewhere else (which may be better or worse).
